### PR TITLE
Overhauls Soap!

### PR DESCRIPTION
**Pull request by Baa14453**
This amazing overhaul adds well-needed quality-of-life enhancements regarding soap.

Moves all soap code out of `code/defines/obj/weapon.dm` and into `code/game/objects/items/weapons/soap.dm`
Removes all soap icons out of `icons/obj/items.dmi` and into `icons/obj/soap.dmi`
Adds a random soap spawner. (Doesn't include syndie soap)
Adds ~~23~~ 19 new soap icons! ~~(Four are not included in the soap spawner because they are based on Earth flags and probably too meta)~~
Replaces non-syndie soap spawners on all maps with the random spawner.
Adds a few extra soap spawns. (Captain's bathroom, sublevel medbay staff showers and some abandoned shower in maintenance)

Added soap spawner in this random shower in Mainlevel maintenance
image

Replaced bars of soap in the Brig with random soap spawners
image

Added spawn spawner in sublevel medical shower
sublevel-medical shower room

Replaced soap in security showers with soap spawner
mainlevel security staff showers

Added a soap spawner to Captain's Shower
mainlevel captain's shower

Replaced bar of soap in psychiatrist locker with soap spawner
interstitial psychiatrists locker

Replaced soap in the surface level showers with a soap spawner 
surface level shower room soaps

Replaced bar of soap on the derelict with a soap spawner
derelict soap

### IC changelog
As part of a station hygiene improvement program, new designs of soap have been delivered.

### CODE REVIEW

**Comment by Baa14453** [comment]
!review required

**Comment by Sindorman** [comment]
I like the change, but I am not sure if moving icons to `soap.dmi` from items would make sense. As other items are all clustered in one file?

**Comment by Baa14453** [comment]
> I like the change, but I am not sure if moving icons to `soap.dmi` from items would make sense. As other items are all clustered in one file?

I thought it might fit better with the other 'weapons'
image

**Comment by Baa14453** [comment]
> I appreciate the change, but please ensure that the descriptions that say "It smells like... " are actually something that you can smell and that makes sense. (A nanotrasen soap that smells like phoron doesn't make a lot of sense)

no worries i will think of some new smells, I based these ones around the Syndie soap which has always smelled 'like fear', but I can adjust them.
btw, the NanoTrasen soap is an original, should I change it anyway?
https://github.com/Aurorastation/Aurora.3/pull/6279/files#diff-c20641f77ac1dd0a57f696ca3efe0741L29

**Comment by VTCobaltblood** [comment]
I am against this change in general, but flag soaps are the most egregious.

**Comment by Baa14453** [comment]
> I am against this change in general, but flag soaps are the most egregious.

The flag soaps were included in the spawner, I just didn't want to waste the sprites, but I've removed them anyway.

File: code/game/objects/items/weapons/soap.dm [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	desc = "A NanoTrasen-brand bar of soap. Smells of phoron."
+	icon_state = "soapnt"
+
+/obj/item/weapon/soap/plant
+	desc = "A green bar of soap. Smells like dirt and plants."
+
+/obj/item/weapon/soap/deluxe
+	icon_state = "soapdeluxe"
+
+/obj/item/weapon/soap/deluxe/New()
```
Comment (Arrow768) [inline code/game/objects/items/weapons/soap.dm]: Use Initialize instead of new

File: code/game/objects/items/weapons/soap.dm [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	name = "Soap"
+	desc = "Smells like butter."
+	icon_state = "golden_soap"
+
+/obj/random/soap/
+	name = "Random Soap"
+	desc = "This is a random soap."
+	icon = 'icons/obj/soap.dmi'
+	icon_state = "soap"
+	item_to_spawn()
```
Comment (Arrow768) [inline code/game/objects/items/weapons/soap.dm]: use absolute paths

File: code/game/objects/items/weapons/soap.dm [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+		return pick(prob(1);/obj/item/weapon/soap, \
+					prob(1);/obj/item/weapon/soap/nanotrasen, \
```
Comment (Arrow768) [inline code/game/objects/items/weapons/soap.dm]: if everything has the same probability, then you dont need to explicitly specify that.
-> pick() and a list of paths inside is going to work fine

File: code/game/objects/items/weapons/soap.dm (hunk 1 above)
Comment (Baa14453) [reply code/game/objects/items/weapons/soap.dm]: Do I just swap out new() with initialise()? That's a legacy thing I didn't make that one.

File: code/game/objects/items/weapons/soap.dm (hunk 2 above)
Comment (Baa14453) [reply code/game/objects/items/weapons/soap.dm]: I added the random spawner to conform with the other random objects inside [random.dm](https://github.com/Baa14453/Aurora.3/blob/SoapOverhaul/code/game/objects/random/random.dm)

Would you prefer:
`/obj/item/weapon/soap/random`

Or:
`/obj/random//weapon/soap`

Or something else maybe, i'm not sure

File: code/game/objects/items/weapons/soap.dm (hunk 3 above)
Comment (Baa14453) [reply code/game/objects/items/weapons/soap.dm]: Ah thanks

File: code/game/objects/items/weapons/soap.dm (hunk 2 above)
Comment (Baa14453) [reply code/game/objects/items/weapons/soap.dm]: Nevermind I was getting confused, like this I think?

```
/obj/item/weapon/soap/random
	name = "Random Soap"
	desc = "This is a random soap."
	icon = 'icons/obj/soap.dmi'
	icon_state = "soap"

/obj/item/weapon/soap/random/item_to_spawn()
		return pick(/obj/item/weapon/soap, \
					/obj/item/weapon/soap/nanotrasen, \
					/obj/item/weapon/soap/deluxe,\
					/obj/item/weapon/soap/space_soap,\
					/obj/item/weapon/soap/space_soap,\
					/obj/item/weapon/soap/water_soap,\
					/obj/item/weapon/soap/fire_soap,\
					/obj/item/weapon/soap/rainbow_soap,\
					/obj/item/weapon/soap/diamond_soap,\
					/obj/item/weapon/soap/uranium_soap,\
					/obj/item/weapon/soap/silver_soap,\
					/obj/item/weapon/soap/brown_soap,\
					/obj/item/weapon/soap/white_soap,\
					/obj/item/weapon/soap/grey_soap,\
					/obj/item/weapon/soap/pink_soap,\
					/obj/item/weapon/soap/purple_soap,\
					/obj/item/weapon/soap/blue_soap,\
					/obj/item/weapon/soap/cyan_soap,\
					/obj/item/weapon/soap/green_soap,\
					/obj/item/weapon/soap/yellow_soap,\
					/obj/item/weapon/soap/orange_soap,\
					/obj/item/weapon/soap/red_soap,\
					/obj/item/weapon/soap/golden_soap
```

Comment (Arrow768) [reply code/game/objects/items/weapons/soap.dm]: Yes that looks good, dont forget to closing )

File: code/game/objects/items/weapons/soap.dm (hunk 1 above)
Comment (Arrow768) [reply code/game/objects/items/weapons/soap.dm]:
```suggestion
/obj/item/weapon/soap/deluxe/Initialize()
	. = ..()
```

Comment (Baa14453) [reply code/game/objects/items/weapons/soap.dm]: Excellent, thanks. I'll test it tomorrow.

**Comment by Baa14453** [comment]
> I like the change, but I am not sure if moving icons to `soap.dmi` from items would make sense. As other items are all clustered in one file?

Sorry, I misread your comment. The icons seem to be all over, I thought it might be a better idea to separate to soap out of items.dmi since there are a few more of them now and so they're a bit easier to find in future.

image

File: code/game/objects/items/weapons/soap.dm (hunk 1 above)
Comment (Baa14453) [reply code/game/objects/items/weapons/soap.dm]: This is working, thanks! :D

**Comment by Baa14453** [comment]
:tada:

**Pull request merged**